tests/ops/op_math/test_condition.py

```diff
@@ -82,7 +82,7 @@ def f(x):
         assert ops[3].base.wires == target_wire
 
         assert len(tape.measurements) == 1
-        assert tape.measurements[0] is terminal_measurement
+        assert tape.measurements[0] == terminal_measurement
```

Comment:
@andrijapau If you scroll up to line 62, the `qml.apply(terminal_measurement)` is the culprit for us needing this change. The update in `transform_dispatcher` is irrelevant here.